ui/src/main/webapp/wizards/sources/stages/discovery.js

```diff
@@ -35,52 +41,100 @@ const discoveryStageDefaults = {
   sourcePort: 8993
 }
 
-const DiscoveryStageView = ({ messages, testSources, setDefaults, configs }) => (
-  <Mount on={() => setDefaults(discoveryStageDefaults)}>
-    <NavPanes backClickTarget='welcomeStage' forwardClickTarget='sourceSelectionStage'>
-      <Title>
-        Discover Available Sources
-      </Title>
-      <Description>
-        Enter connection information to scan for available sources on a host.
-      </Description>
-      <div style={{ width: 400, position: 'relative', margin: '0px auto', padding: 0 }}>
-        <Hostname
-          id='sourceHostName'
-          label='Hostname'
-          autoFocus />
-        <Port
-          id='sourcePort'
-          label='Port'
-          errorText={(configs.sourcePort < 0) ? 'Port is not in valid range.' : undefined} />
-        <Input
-          id='sourceUserName'
-          label='Username (optional)'
-          errorText={(isBlank(configs.sourceUserName) && !isEmpty(configs.sourceUserPassword)) ? 'Password with no username.' : undefined} />
-        <Password
-          id='sourceUserPassword'
-          label='Password (optional)'
-          errorText={(!isBlank(configs.sourceUserName) && isEmpty(configs.sourceUserPassword)) ? 'Username with no password.' : undefined} />
-        {messages.map((msg, i) => <Message key={i} {...msg} />)}
-      </div>
-
-      <ActionGroup>
-        <Action
-          primary
-          label='Check'
-          disabled={isBlank(configs.sourceHostName) ||
-          configs.sourcePort < 0 ||
-          isBlank(configs.sourceUserName) !== isEmpty(configs.sourceUserPassword)}
-          onClick={() => testSources('sources', 'sourceSelectionStage', 'discoveryStage')} />
-      </ActionGroup>
-    </NavPanes>
-  </Mount>
-)
+const DiscoveryStageView = ({ messages, testSources, setDefaults, configs, discoveryType, setDiscoveryType }) => {
+  const nextShouldBeDisabled = () => {
+    // check to see if username/password entry is mismatched
+    if (isBlank(configs.sourceUserName) !== isEmpty(configs.sourceUserPassword)) {
```

Comment:
This makes sure that the user either gave no username/password or that they gave both. If they gave a user and no password or a password and no user, they aren't allowed to submit and there is an error message on the respective box explaining that they forgot their password or that they forgot their username.